drivers/platform/x86/asus-wmi.c

```diff
@@ -1745,6 +1754,14 @@ static void asus_wmi_notify(u32 value, void *context)
 		}
 	}
 
+	if (is_kbd_led_event(code)) {
+		if (code == NOTIFY_KBD_BRTDWN)
+			kbd_led_set(&asus->kbd_led, asus->kbd_led_wk > 0 ? --asus->kbd_led_wk : LED_OFF);
+		else
+			kbd_led_set(&asus->kbd_led, ++asus->kbd_led_wk);
```

Comment:
I think you introduce a behaviour change here (compared to the current situation handled in userspace): If I press the brightness up key 10 times and then I press brightness down once, then the brightness down key will have no effect.
I think you need to clamp kbd_led_wk within the usable range?

Reply:
I decided to make some change in kbd_led_set to compare under the same data type integer instead of enum led_brightness. Also, modify the commit message for an explanation. Is it better?

Reply:
I had not seen that kbd_led_set was assigning a new (correctly bounded) value to kbd_led_wk, so now I see that my last comment was not correct.

The reason for my confusion here was that you are changing the value of kbd_led_wk here in this function, in addition to it being changed in kbd_led_set. I think you should limit the value change just to one place - kbd_led_set.
e.g. here you can change `--asus->kbd_led_wk` to `asus->kbd_led_wk - 1`

And please correct me if I'm wrong but I don't think the (int) casts you added have any effect. enum values are already type int.

Reply:
I also thought the enum should be type int too. However, seems not. When the value (asus->kbd_led_wk) comes to -1, the `if (value > asus->kbd_led.max_brightness)` return true and `if (value < 0)` also return false. That's why the Fn+F3 becomes cylic, 3 -> 2 -> 1 -> 0 -> -1 ( set to 3 by kbd_led_set) -> 2 -> 1. After I changed to `((int)va.lue > (int)asus->kbd_led.max_brightness)`, the `if (value < 0)` would be executed but always return fase, then each Fn+F3 would be 3 -> 2 -> 1 -> 0 -> -1 -> -2 -> -3 ->..... That's why I have to do casting for these 2 `if` condition. I'll paste the assembly code of `kbd_led_set` if necessary.

Reply:
Looking closer, looks like the compiler is free to decide which type to use for the enum, and in this case it must be picking an unsigned type.